benchmark/vochain_benchmark_test.go

```diff
@@ -273,19 +194,66 @@ func voteBench(b *testing.B, cl *client.Client, s *ethereum.SignKeys,
 		b.Fatal(err)
 	}
 	// sending submitEnvelope request
-	resp = doRequest("submitRawTx", nil)
-	log.Infof("response: %s", resp.String())
+	resp := doRequest("submitRawTx", nil)
+	if !resp.Ok {
+		b.Errorf("error submitting vote for %s: %s", s.AddressString(), resp.String())
+	}
+}
 
-	// check vote added
-	/*	reset(req)
-		req.ProcessID = processID
-		req.Nullifier = vochain.GenerateNullifier(s.Address(), processID)
-		for {
-			resp = doRequest("getEnvelopeStatus", nil)
-			if *resp.Registered {
-				break
+func waitProcess(b *testing.B, cl *client.Client, processID []byte) {
+	log.Info("waiting for the process to start")
+	req := &api.APIrequest{}
+	reset(req)
+	doRequest := cl.ForTest(b, req)
+	req.ProcessID = processID
+	failures := 20
+	for {
+		procInfo := doRequest("getProcessInfo", nil)
+		if procInfo.Ok {
+			if procInfo.Process.Status == int32(models.ProcessStatus_READY) {
+				height := doRequest("getBlockHeight", nil)
+				if *height.Height >= procInfo.Process.StartBlock {
+					break
+				}
 			}
-			time.Sleep(time.Second * 2)
 		}
-	*/
+		failures--
+		if failures == 0 {
+			b.Fatalf("processID %s should be started by now", hexProcessID)
+		}
+		time.Sleep(time.Second)
+	}
+}
+
+func createDistributedCensusAndProcess(b *testing.B, cl *client.Client, dvoteServer testcommon.DvoteAPIServer, keySet []*ethereum.SignKeys, entityID []byte) (censusRoot []byte, censusURI string, processID []byte) {
```

Comment:
The signature of this function is too long, please split it like: 

```golang
func createDistributedCensusAndProcess(b *testing.B,
    cl *client.Client,
    dvoteServer testcommon.DvoteAPIServer,
    keySet []*ethereum.SignKeys,
    entityID []byte) (censusRoot []byte, censusURI string, processID []byte) {
```
Also as commented before I think we can return an error and do the Fatal in the upper layer

Reply:
Btw, I personally do not like a lot this function, `DistributedCensus` does not add much information, as the census are normally distributed under the Vocdoni approach. And as we already have a create process and create census function I think is fine to keep them separated. But if this duplicated a lot of code maybe is fine.
If you decide to keep it please change the name to `createCensusAndProcess`

Reply:
The name is like that so in the future one won't ask "why doesn't BenchmarkVochainCSP" use this - but yeah it is an ugly name.

I'll rename it, but the function saves a lot of code duplication as you can see. Setup code is already long enough as it is, this is an easy way to keep it somewhat legible.

Reply:
I decided to keep it because the aim was always to keep the brain free while reading the higher level code, not code tidiness per se.